### DISCUSSION TITLE
Is there a way to convert metric units into a bigger ones automatically?

### DISCUSSION THREAD

**Discussion by zawnk** [mealie-recipes/mealie#3336] [Q&A] [answered]
Hi,

I was just wondering if I'm missing a way to 'link' units or otherwise make it possible for metric units to be converted automatically. For example:

I have a recipe that requires `600g of flour` and when I double the recipe it says `1200g of flour`. Is there a way to have it displayed as `1.2kg of flour`?

Just wanting to make sure I don't miss something obvious :) Thanks!

**Comment by Kuchenpirat** [comment, marked as answer]
There is not. Units are currently user provided (the seeded units do not get handled differently) so it is impossible for us to have a relation between units with the current system.

**Reply by zawnk** [reply]
Thanks for the quick response! Maybe down the line there could be a functionality to 'link' units and have the user give a ratio to convert between those two. But totally low prio. Thanks again.

**Comment by Ronaldvr** [comment]
Ah the imperial unit methodology strikes again! If and when you use imperial this is a nightmare however if/when you use SI units this is an easy thing to implement. And even if mealie would simply *use* SI unites _internally_ and only *convert* to imperial when requested the base arithmetic would still be very easy: just multiples of 10/100/1000 and convert on request. (And in fact most imperial units are actually tied to SI: https://en.wikipedia.org/wiki/Imperial_units#United_Kingdom
See also https://github.com/mealie-recipes/mealie/discussions/3881
